src/fluree/db/datatype.cljc

```diff
@@ -47,6 +47,35 @@
    "http://www.w3.org/1999/02/22-rdf-syntax-ns#langString" const/$rdf:langString})
 
 
+(def time-types
+  #{const/$xsd:date
+    const/$xsd:dateTime
+    const/$xsd:time})
+
+#?(:clj (def ^DateTimeFormatter xsdDateTimeFormatter
+          (DateTimeFormatter/ofPattern "uuuu-MM-dd'T'HH:mm:ss.SSSSSSSSS[XXXXX]")))
+
+#?(:clj (def ^DateTimeFormatter xsdTimeFormatter
+          (DateTimeFormatter/ofPattern "HH:mm:ss.SSSSSSSSS[XXXXX]")))
+
+#?(:clj (def ^DateTimeFormatter xsdDateFormatter
+          (DateTimeFormatter/ofPattern "uuuu-MM-dd[XXXXX]")))
+
+(defn serialize-flake-value
+  "Flakes with time types will have time objects as values.
+  We need to serialize these into strings that will be successfully re-coerced into
+  the same objects upon loading."
+  [val dt]
+  (uc/case (int dt)
+    const/$xsd:dateTime #?(:clj (.format xsdDateTimeFormatter val)
+                           :cljs (.toJSON val))
+    const/$xsd:date      #?(:clj (.format xsdDateFormatter val)
+                            :cljs (.toJSON val))
+    const/$xsd:time #?(:clj (.format xsdTimeFormatter val)
+                       :cljs (.toJSON val))
+    val))
```

Comment:
Serialization doesn't _quite_ seem like it belongs in this ns, but it also felt weird to pull this over into `commit` from `serde.json`, which is otherwise unrelated to commit-writing.  If there's a better place than this, I'm happy to move it

Reply:
I see why you moved it, though it does feel weird to have the serialization code in this namespace instead of serde.

Reply:
Oops, I commented without refreshing first to see if there were more comments. : )